pkg/util/xorm/sequence.go

```diff
@@ -3,65 +3,142 @@ package xorm
 import (
 	"context"
 	"database/sql"
-	"errors"
 	"fmt"
+	"sync"
 )
 
+// batchState represents the state of a sequence batch
+type batchState struct {
+	nextValue        int64
+	lastValueInBatch int64
+	mu               sync.Mutex
```

Comment:
nit: can we move `mu` on top of variables it protects? (mutex-hat)

same comment in `sequenceGenerator`